### PR TITLE
Add master_external_loadbalancer config param to the config reference.

### PR DESCRIPTION
**Pull request by ahairape**
## Description
This PR adds the configuration parameter `master_external_loadbalancer` to the configuration reference sections for DC/OS EE 1.10 and 1.11.
https://jira.mesosphere.com/browse/COPS-2114

## Urgency
- [ ] Blocker 
- [ x] High
- [ ] Medium

### CODE REVIEW

**Comment by ahairape** [comment]
@jgehrcke could you please comment on the following?

- Does the configuration parameter `master_external_loadbalancer` need to be added also to the configuration references (1.10 and 1.11) of DC/OS OSS?
- I have put this option in the subsection *Cluster Setup* of the configuration reference. Do you think any other subsection (*Networking*, *Performance and Tuning* or *Security and Authentication*) could be more appropriate?

Thank you!

**Comment by jgehrcke** [comment]
@mhrabovcin  can you please review that? :) thanks.

Is the value type restricted to DNS name or can this also be an IP address?

File: pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ This topic provides configuration parameters available for [DC/OS Enterprise](ht
 | [gpus_are_scarce](#gpus-are-scarce)                                   | Indicates whether to treat GPUs as a scarce resource in the cluster. |
 | [ip_detect_public_filename](#ip-detect-public-filename)               | The IP detect file to use in your cluster.  |
 | [master_discovery](#master-discovery)                                 | (Required) The Mesos master discovery method.         |
+| [master_external_loadbalancer](#master-external-loadbalancer)         | The DNS name for the load balancer.         |
```
Comment (mhrabovcin) [inline pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md]: As @jgehrcke spotted the `master_external_loadbalancer` value can be also an IP address.

https://github.com/mesosphere/dcos-enterprise/blob/1.11/packages/bootstrap/extra/dcos_internal_utils/bootstrap.py#L2053-L2057

Comment (ahairape) [reply pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md]: fixed

File: pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md [hunk 2]
```diff
@@ -389,6 +390,9 @@ Indicates whether the master DNS port is open. An open master DNS port listens p
 *  `master_dns_bindall: 'true'` The master DNS port is open. This is the default value.
 *  `master_dns_bindall: 'false'` The master DNS port is closed.
 
+### master_external_loadbalancer
+The DNS name or IP address for the load balancer.  If specified, this name is included in [DC/OS certificates](/1.10/security/ent/tls-ssl/).
```
Comment (mhrabovcin) [inline pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md]: There are 2 spaces here `balancer.  If`

File: pages/1.11/installing/ent/custom/configuration/configuration-parameters/index.md [hunk 3]
```diff
@@ -439,6 +440,9 @@ Indicates whether the master DNS port is open. An open master DNS port listens p
 *  `master_dns_bindall: 'true'` The master DNS port is open. This is the default value.
 *  `master_dns_bindall: 'false'` The master DNS port is closed.
 
+### master_external_loadbalancer
+The DNS name or IP address for the load balancer.  If specified, this name is included in [DC/OS certificates](/1.11/security/ent/tls-ssl/).
```
Comment (mhrabovcin) [inline pages/1.11/installing/ent/custom/configuration/configuration-parameters/index.md]: There are 2 spaces here `balancer.  If`

Comment (ahairape) [reply pages/1.11/installing/ent/custom/configuration/configuration-parameters/index.md]: fixed

**Comment by mhrabovcin** [comment]
> Does the configuration parameter `master_external_loadbalancer` need to be added also to the configuration references (1.10 and 1.11) of DC/OS OSS?

I haven't found ay use of `master_external_loadbalancer` in OSS version.

File: pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md [hunk 4]
```diff
@@ -389,6 +390,9 @@ Indicates whether the master DNS port is open. An open master DNS port listens p
 *  `master_dns_bindall: 'true'` The master DNS port is open. This is the default value.
 *  `master_dns_bindall: 'false'` The master DNS port is closed.
 
+### master_external_loadbalancer
+The DNS name or IP address for the load balancer. If specified, this name is included in [DC/OS certificates](/1.10/security/ent/tls-ssl/).
```
Comment (jgehrcke) [inline pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md]: > this name is included

"this name" is weird with the previous "The DNS name or IP address".

Maybe "this is included as subject alternative name..."?

> included in [DC/OS certificates]

In all of them? Should we be more precise?

Comment (ahairape) [reply pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md]: Added a bit more specific info.

File: pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md [hunk 5]
```diff
@@ -389,6 +390,9 @@ Indicates whether the master DNS port is open. An open master DNS port listens p
 *  `master_dns_bindall: 'true'` The master DNS port is open. This is the default value.
 *  `master_dns_bindall: 'false'` The master DNS port is closed.
 
+### master_external_loadbalancer
+The DNS name or IP address for the load balancer. If specified, this is included as subject alternative name in the [DC/OS certificate](/1.10/security/ent/tls-ssl/) of the host on which the load balancer is running.
```
Comment (jgehrcke) [inline pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md]: > of the host on which the load balancer is running.

This is not correct.

It makes *Admin Router on the master nodes* present an X.509 certificate including the subject alternative name corresponding to the IP address / DNS name of the level 4 load balancer in front of it (this is meant to be used for a non-TLS-terminating level **4** load balancer running on a host/hosts that is/are *not* part of DC/OS, in front of the master nodes, routing _external_ traffic to Admin Router on the master nodes).

Example is our AWS deployment. For example, `adyatlov-elasticl-17kqbudnuzk0y-682688633.us-west-2.elb.amazonaws.com` is a DNS name that represents a TCP level load balancer in front of the master nodes, whereas when the browser accesses `https://adyatlov-elasticl-17kqbudnuzk0y-68268863.....` the TLS handshake is directly performed with Admin Router running on the master node that the TCP connection was routed to.

Comment (ahairape) [reply pages/1.10/installing/ent/custom/configuration/configuration-parameters/index.md]: Thank you very much for the explanation. I have modified the text accordingly.

**Pull request merged**